src/server/mod.rs

```diff
@@ -2664,3 +2478,14 @@ fn calculate_relative_teleport(flag: TeleportFlag, flags: u8, base: f64, val: f6
         base + val
     }
 }
+
+pub struct Network {
```

Comment:
This doesn't seem very well fledged out, why not put the version inside the Conn directly or pass it when calling connect?

Reply:
Ah, I didn't know Conn contained protocol_version. I've switch a few of the generalized protocol operations to use the version from Conn instead.